examples/platform/silabs/efr32/rs911x/wfx_rsi_host.cpp

```diff
@@ -0,0 +1,425 @@
+ * @brief
+ * Called from ConnectivityManagerImpl.cpp - to enable the device
+ * Create the RSI task and let it deal with life.
+ * @param[in]  None
+ * @return  Returns SL_STATUS_OK if successful,
+ *          SL_STATUS_FAIL otherwise
+ ***********************************************************************/
+sl_status_t wfx_wifi_start(void)
+{
+    if (wfx_rsi.dev_state & WFX_RSI_ST_STARTED)
+    {
+        return SL_STATUS_OK;
+    }
+    wfx_rsi.dev_state |= WFX_RSI_ST_STARTED;
+    /*
+     * Create the Wifi driver task
+     */
+    wfx_rsi.wlan_task = xTaskCreateStatic(wfx_rsi_task, "wfx_rsi", WFX_RSI_TASK_SZ, NULL, WLAN_DRIVER_TASK_PRIORITY,
+                                          wfxRsiTaskStack, &wfxRsiTaskBuffer);
+
+    if (NULL == wfx_rsi.wlan_task)
+    {
+        return SL_STATUS_FAIL;
+    }
+    return SL_STATUS_OK;
+}
+
+/*********************************************************************
+ * @fn  void wfx_enable_sta_mode(void)
+ * @brief
+ *      driver enable the STA mode
+ * @param[in]  None
+ * @return   None
+ ***********************************************************************/
+void wfx_enable_sta_mode(void)
+{
+    wfx_rsi.dev_state |= WFX_RSI_ST_STA_MODE;
+}
+
+/*********************************************************************
+ * @fn  bool wfx_is_sta_mode_enabled(void)
+ * @brief
+ *      driver enabled the STA mode
+ * @param[in]  None
+ * @return   mode
+ ***********************************************************************/
+bool wfx_is_sta_mode_enabled(void)
+{
+    bool mode;
+    mode = !!(wfx_rsi.dev_state & WFX_RSI_ST_STA_MODE);
+    return mode;
+}
+
+/*********************************************************************
+ * @fn  wfx_get_wifi_mac_addr(sl_wfx_interface_t interface, sl_wfx_mac_address_t * addr)
+ * @brief
+ *      get the wifi mac address
+ * @param[in]  Interface:
+ * @param[in]  addr : address
+ * @return
+ *       None
+ ***********************************************************************/
+void wfx_get_wifi_mac_addr(sl_wfx_interface_t interface, sl_wfx_mac_address_t * addr)
+{
+    sl_wfx_mac_address_t * mac;
+
+#ifdef SL_WFX_CONFIG_SOFTAP
+    mac = (interface == SL_WFX_SOFTAP_INTERFACE) ? &wfx_rsi.softap_mac : &wfx_rsi.sta_mac;
+#else
+    mac = &wfx_rsi.sta_mac;
+#endif
+    *addr = *mac;
+    SILABS_LOG("%s: %02x:%02x:%02x:%02x:%02x:%02x", __func__, mac->octet[0], mac->octet[1], mac->octet[2], mac->octet[3],
+               mac->octet[4], mac->octet[5]);
+}
+
+/*********************************************************************
+ * @fn  void wfx_set_wifi_provision(wfx_wifi_provision_t *cfg)
+ * @brief
+ *      Driver set the wifi provision
+ * @param[in]  cfg:  wifi configuration
+ * @return
+ *       None
+ ***********************************************************************/
+void wfx_set_wifi_provision(wfx_wifi_provision_t * cfg)
+{
+    SILABS_LOG("%s: SSID: %s", __func__, &wfx_rsi.sec.ssid[0]);
+
+    wfx_rsi.sec = *cfg;
```

Comment:
No nullptr verification, should return immediatly if the struc is a `nullptr`

Reply:
Null pointer checked.